lib/rubocop/server/cache.rb

```diff
@@ -41,6 +43,14 @@ def project_dir_cache_key
           @project_dir_cache_key ||= project_dir[1..].tr('/', '+')
         end
 
+        def restart_key
+          lockfile_path = LOCKFILE_NAMES.map do |lockfile_name|
+            Pathname(project_dir).join(lockfile_name)
+          end.find(&:exist?)
+
+          Digest::SHA1.hexdigest(lockfile_path&.read || RuboCop::Version::STRING)
```

Comment:
Does it make sense to `read.grep(/rubocop/)` to avoid restarting on unrelated updates?

Reply:
Unrelated is a bit fuzzy here. For example, if I conservatively update the `parser` gem because a new ruby version released I'd still want it to restart rubocop.

Personally I think its safer to keep it like that, having it run more closely to how it would do without server mode.

Reply:
Yes. It is prioritize that the environment can be aligned even when updating dependent gems that don't match `/rubocop/`, to prevent user confusion. Especially since it is not possible to know in advance which gems 3rd party gem depends on. Also, since `bundle update` occurs less frequently than `rubocop` is run, it has been decided that updating the entire Gemfile.lock to SHA1 is a trade-off that makes sense.